### PR TITLE
Add ninny-json ADT benchmark

### PR DESCRIPTION
**Pull request by kag0**
Thought I'd see how my library stacks for the simple ADT benchmark. 

Serialization is implemented with extreme naivete, so I'm interested to see if a switch to jsoniter would be beneficial (de serialization already uses jawn).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,4 +28,7 @@ TAGS
 
 # Visual Studio Code
 .metals
-.bloop
\ No newline at end of file
+.bloop
+.vscode/settings.json
```
Comment (kag0) [inline .gitignore:32]: let me know if these ignores are unwanted, they're just what I found in the diff after making some edits in vs code

**Comment by plokhotnyuk** [comment]
Why not? 

Both play-json and circe have boosters for parsing/serialization to/from their in-memory JSON representations and decoding/encoding of some types like `java.time.*` based on jsoniter-scala. 

You can create 2 sets of benchmarks for current implementation and for boosted. 

There is only one requirement they should pass existing benchmark tests.

File: jsoniter-scala-benchmark/jvm/src/test/scala/com/github/plokhotnyuk/jsoniter_scala/benchmark/ADTWritingSpec.scala [hunk 2]
```diff
@@ -13,6 +13,7 @@ class ADTWritingSpec extends BenchmarkSpecBase {
       toString(b.jacksonScala()) shouldBe b.jsonString1
       toString(b.jsoniterScala()) shouldBe b.jsonString1
       toString(b.preallocatedBuf, 0, b.jsoniterScalaPrealloc()) shouldBe b.jsonString1
+      toString(b.ninnyJson()) shouldBe b.jsonString1
```
Comment (kag0) [inline jsoniter-scala-benchmark/jvm/src/test/scala/com/github/plokhotnyuk/jsoniter_scala/benchmark/ADTWritingSpec.scala:16]: why do some of these tests use `b.jsonString1` and others use `b.jsonString2`?

**Comment by kag0** [comment]
the test failures seem to do with not being able to resolve imports. it "works on my machine" when running `sbt jsoniter-scala-benchmarkJVM/test`, is there anywhere I might be missing a dependency or difference between CI and what I ran?

**Comment by plokhotnyuk** [comment]
> the test failures seem to do with not being able to resolve imports. it "compiles on my machine" when running `sbt jsoniter-scala-benchmarkJVM/test`, is there anywhere I might be missing a dependency or difference between CI and what I ran?

Do you have your library published somewhere to be used by adding its artifact(s) to [this list of benchmark dependencies](https://github.com/plokhotnyuk/jsoniter-scala/blob/master/build.sbt#L226)?

**Comment by kag0** [comment]
ah, duh
I had skipped checking in `build.sbt` because I was fiddling with the scala version to make semanticdb happy

**Comment by plokhotnyuk** [comment]
@kag0 All tests passed for now. 

Would you like to add more benchmarks for your library in this PR? 

Most interesting are benchmarks for real-world message samples: GeoJSON, GitHub API, GoogleMaps API, OpenRTB, and Twitter API.

**Comment by kag0** [comment]
I'd like to add some of the real-world benchmarks. But I'd prefer to do it in a follow-up PR if that's OK

**Pull request merged**